Geometry_Engine/Create/ShapeProfile.cs

```diff
@@ -148,6 +256,16 @@ public static TSectionProfile TSectionProfile(double height, double width, doubl
 
         public static GeneralisedTSectionProfile GeneralisedTSectionProfile(double height, double webThickness, double leftOutstandWidth, double leftOutstandThickness, double rightOutstandWidth, double rightOutstandThickness, bool mirrorAboutLocalY = false)
         {
+            if (height <= leftOutstandThickness || height <= rightOutstandThickness)
+            {
+                Engine.Reflection.Compute.RecordError("The ratio between inputs makes section inconceivable");
+                return null;
+            }
+            if (height <= 0 || webThickness <= 0 || leftOutstandThickness <= 0 || leftOutstandThickness <= 0 || rightOutstandThickness < 0 || rightOutstandWidth <= 0)
```

Comment:
Some inconsistencies here for righthand vs lefthand side. 

Think the check needs to be that outstands widths can be 0 and that if outstandwidths are non-0, then the thickness need to be above 0. Basically, need to avoid being able to make these kinds of sections:

![image](https://user-images.githubusercontent.com/22005920/66906586-420c0780-f008-11e9-9945-b76a79cb9b84.png)

but this should still be allowed:

![image](https://user-images.githubusercontent.com/22005920/66906662-5b14b880-f008-11e9-8f56-49a651a0563e.png)